builder/rootfs/bin/boot

```diff
@@ -74,6 +88,27 @@ done
 docker build -t deis/slugbuilder /usr/local/src/slugbuilder/
 docker build -t deis/slugrunner /usr/local/src/slugrunner/
 
+function gen_host_keys {
+	# generate the keys, then set them up in etcd if not already set
+	/usr/bin/ssh-keygen -A
```

Comment:
I see. But without comments to that effect or a close read of the code to notice that `etcdctl mk` is used, the code looks incorrect. Let's refactor it to make it more readable. I think branching logic is fine here to pay for clarify. :+1: